crates/payload/builder/src/payload.rs

```diff
@@ -145,9 +159,22 @@ impl PayloadBuilderAttributes {
     /// Creates a new payload builder for the given parent block and the attributes.
     ///
     /// Derives the unique [PayloadId] for the given parent and attributes
-    pub fn new(parent: B256, attributes: PayloadAttributes) -> Self {
+    pub fn try_new(parent: B256, attributes: PayloadAttributes) -> Result<Self, DecodeError> {
+        #[cfg(not(feature = "optimism"))]
         let id = payload_id(&parent, &attributes);
 
+        #[cfg(feature = "optimism")]
+        let (id, transactions) = {
+            let transactions = attributes
+                .transactions
+                .as_ref()
+                .unwrap_or(&Vec::default())
```

Comment:
this should work
```suggestion
                .as_deref()
                .unwrap_or(&[])
```

Reply:
Addressed in https://github.com/anton-rs/op-reth/pull/128